src/ansys/systemcoupling/core/charts/chart_datatypes.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Optional
+
+# Common intermediate metadata and data types for chart series information.
+# All series are assumed to belong to an interface and a data transfer within the interface
+
+# Idea is that data should be processed into this format from whatever raw source (CSV file,
+# streamed data, etc.) and the actual charts will be built from this.
+
+
+class SeriesType(Enum):
+    CONVERGENCE = 1
+    SUM = 2
+    WEIGHTED_AVERAGE = 3
+
+
+@dataclass
+class TransferSeriesInfo:
+    """Information about chart series data associated with a data transfer.
```

Comment:
```suggestion
    """Stores information about the chart series data associated with a data transfer.
```
Descriptions for classes start with a verb ending in "s" or "es"

Reply:
I am not sure about this guideline. In OOD, the usual (simplified) model is to regard classes as nouns, and methods and functions as verbs. This implies that the description should be about what the class represents. My understanding seems to be consistent with what the Google Python style guide says.

Furthermore, Google also cautions against repeating unnecessary information, such as that the class is a class. Given that a `@dataclass` is by definition a structure for storing data, it seems redundant to have to prefix every `@dataclass` docstring with "Stores ...".

Reply:
Hmm. Your point is taken. Back several years ago, working with Alex K., we determined we'd start class descriptions with a verb ending in an "s" or "es" rather than describe it as a noun. The only noun description we use is for a property. What is OOD? I'd like to investigate a bit more! Thanks!

Reply:
OOA/OOD is Object Oriented Analysis/Design. It's the paradigm supported by languages that have the concept of classes. There is a small summary of a "natural language analysis" technique on [this page](https://cs.fsu.edu/~myers/cop3331/notes/analysis1.html) and some slides [here](https://www.se.rit.edu/~swen-262/slides/Nouns%20and%20Verbs.pdf) that go into more detail.

Reply:
I am going to mark this conversation as resolved. Even though there could be some further discussion to be had, it is probably beyond the scope of this PR. Also, this is internal code doc that won't appear in the published API docs and relates to code that is likely to be revisited and revised soon anyway.